libraries/fc/src/crypto/elliptic_common.cpp

```diff
@@ -168,11 +173,48 @@ namespace fc { namespace ecc {
         return (fp[0] << 24) | (fp[1] << 16) | (fp[2] << 8) | fp[3];
     }
 
-    bool public_key::is_canonical( const compact_signature& c ) {
-        return !(c.data[1] & 0x80)
-               && !(c.data[1] == 0 && !(c.data[2] & 0x80))
-               && !(c.data[33] & 0x80)
-               && !(c.data[33] == 0 && !(c.data[34] & 0x80));
+    bool is_fc_canonical( const compact_signature& c )
+    {
+      return !(c.data[1] & 0x80)
+          && !(c.data[1] == 0 && !(c.data[2] & 0x80))
+          && !(c.data[33] & 0x80)
+          && !(c.data[33] == 0 && !(c.data[34] & 0x80));
+    }
+
+    bool is_bip_0062_canonical( const compact_signature& c )
+    {
+       constexpr boost::multiprecision::uint256_t n_2 =
+          0x7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF5D576E7357A4501DDFE92F46681B20A0_cppui256;
+
+       boost::multiprecision::uint256_t sig = 0;
+
+       // boost endian conversions are only supported for native c++ types,
+       // so we need to convert in 64 bit words
+       for( size_t i = 0; i < 4; i++ )
+       {
+          sig <<= 64;
+          sig += boost::endian::big_to_native( *( uint64_t* )( c.data + 33 + ( i * 8 ) ) );
```

Comment:
I'm not convinced all this complexity is necessary. Since the numbers are both guaranteed to be big-endian and fixed-length, a simple `memcmp` should be all that's necessary to determine which is the larger number.